src/components/icon/index.jsx

```diff
@@ -9,10 +9,25 @@ import './index.css';
  * @returns A Generic Icon
  */
 function Icon(props) {
-	const { iconName, iconColor, iconBackgroundColor, iconSize, iconBordered } = props;
-	return (
-        <SemanticIcon className={"icon " + iconColor + " " + iconBackgroundColor} name={iconName} size={iconSize} bordered={iconBordered}/>
-	);
+	const { iconName, iconColor, iconBackgroundColor, iconSize, iconBordered, iconHref } = props;
+	const handleClick = (href) => {
+		if(href.includes("mailto:")) {
+			window.location.href = href;
+		}
+		else {
+			window.open(href, '_blank');
+		}
+	}
+	if (iconHref !== undefined) {
+		return (
+			<SemanticIcon className={"icon social-icon " + iconColor + " " + iconBackgroundColor} name={iconName} size={iconSize} bordered={iconBordered} onClick={() => handleClick(iconHref ? iconHref : "")}/>
+		);
+	}
+	else {
```

Comment:
same here

Reply:
This if/else statement is used to differentiate an icon used for social media (i.e. YouTube) from one that is only used as an icon, such as the checkmark seen in CW-22